src/components/sidebar/sidebar.jsx

```diff
@@ -0,0 +1,143 @@
+import React, {
+	createContext,
+	useContext,
+	useState,
+	useRef,
+	useEffect,
+} from 'react';
+import { cn } from '@/utilities/functions';
+import { PanelLeftClose, PanelLeftOpen } from 'lucide-react';
+import Tooltip from '../tooltip';
+const SidebarContext = createContext();
+
+const Sidebar = ( {
+	children,
+	className,
+	onCollapseChange,
+	collapsible = true,
+	screenHeight = true,
+	borderOn = true,
+	...props
+} ) => {
+	const sideBarRef = useRef( null );
+	const [ isCollapsed, setIsCollapsed ] = useState( () => {
+		// eslint-disable-next-line no-undef
```

Comment:
We can add `localStorage` as a global variable in the eslint config instead of disabling it.

Reply:
> We can add `localStorage` as a global variable in the eslint config instead of disabling it.

Added